tabular/src/autogluon/tabular/learner/default_learner.py

```diff
@@ -65,6 +66,29 @@ def _fit(self, X: DataFrame, X_val: DataFrame = None, X_unlabeled: DataFrame = N
         logger.log(20, f'Operating System:   {platform.system()}')
         logger.log(20, f'Platform Machine:   {platform.machine()}')
         logger.log(20, f'Platform Version:   {platform.version()}')
+        try:
+            # TODO: Make this logic smarter, incorporate training data size and potentially models to train into the logic to define the recommended disk space.
+            #  For example, `best_quality` will require more disk space than `medium_quality`, and HPO would require additional disk space.
+            disk_stats = ResourceManager.get_disk_usage(path=self.path)
+            disk_free_gb = disk_stats.free / 1e9
+            disk_total_gb = disk_stats.total / 1e9
+            disk_proportion_avail = disk_stats.free / disk_stats.total
+            disk_log_extra = ''
+            disk_free_gb_warning_threshold = 10
+            disk_verbosity = 20
+            if disk_free_gb <= disk_free_gb_warning_threshold:
+                disk_log_extra += f'\n\tWARNING: Available disk space is low and there is a risk that ' \
+                                  f'AutoGluon will run out of disk during fit, causing an exception. ' \
+                                  f'\n\tWe recommend a minimum available disk space of {disk_free_gb_warning_threshold} GB, ' \
+                                  f'and large datasets may require more.'
+                disk_verbosity = 30
+            logger.log(disk_verbosity,
+                       f'Disk Space Avail:   {disk_free_gb:.2f} GB / {disk_total_gb:.2f} GB '
+                       f'({disk_proportion_avail * 100:.1f}%){disk_log_extra}')
+        except Exception as e:
```

Comment:
Its best to catch as specific of an exception as possible. This would allow us to make the following log statement if the issue is indeed related to the `disk_usage` call, but if another error is raised in this try block, we don't accidentally hide if from the user. If you know of a more specific exception, it could be good to use it here.

Reply:
It is unclear what exception it would raise, as it would require me being able to simulate some strange unknown environment where this logic doesn't work correctly, and the shutil code does not make it clear what exception would be raised in these strange situations. Because of this, it is safest to have a generic exception catch to avoid completely breaking AutoGluon in these more niche scenarios that I don't know about. I'm unsure if these scenarios even exist, but it is very hard to rule out the possibility. (For example, running in the browser, running in colab, running in kaggle, running in docker, running on windows, running on specialized private hardware, running in a fully-in-memory disk environment, etc.)

Reply:
Added in-line comment explaining this in the code